### PR TITLE
feat: added layout structure

### PR DESCRIPTION
**Pull request by azawert**

### CODE REVIEW

File: src/widget/header/ui/header.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    Switch,
+    rem,
+    useMantineTheme,
+} from '@mantine/core'
+
+import { IconSun, IconMoonStars, IconWorld } from '@tabler/icons-react'
+export const Header: FC = () => {
+    const theme = useMantineTheme()
+
+    const sunIcon = (
```
Comment (suveiko) [inline src/widget/header/ui/header.tsx]: put all the icons in one object and pick them up by key

`const icons = {
        sun: (
            <IconSun
                style={{ width: rem(16), height: rem(16) }}
                stroke={2.5}
                color={theme.colors.yellow[4]}
            />
        ),
        moon: (
            <IconMoonStars
                style={{ width: rem(16), height: rem(16) }}
                stroke={2.5}
                color={theme.colors.blue[6]}
            />
        ),
        world: (
            <IconWorld
                style={{ width: rem(18), height: rem(18) }}
                stroke={1.5}
            />
        ),
    }`
    
    like this

File: src/shared/ui/layout/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { AppShell } from '@mantine/core'
+import { FC, ReactNode } from 'react'
+import { Header } from '../../../widget/header'
+import { Navbar } from '../../../widget/navbar'
+
+const Layout: FC<{ children: ReactNode }> = ({ children }) => {
```
Comment (suveiko) [inline src/shared/ui/layout/layout.tsx]: remove unnecessary type `{ children: ReactNode }`

File: src/App.tsx [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import { MantineProvider } from '@mantine/core'
 import { StoreProvider } from './app/provider/store'
+import Layout from './shared/ui/layout/layout'
 
 function App() {
     return (
         <StoreProvider>
             <MantineProvider>
-                <div></div>
+                {/* main content */}
```
Comment (remadyt) [inline src/App.tsx]: remove comment

File: src/widget/header/ui/header.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    Flex,
+    Group,
+    Image,
+    Select,
+    Switch,
+    rem,
+    useMantineTheme,
+} from '@mantine/core'
+
+import { IconSun, IconMoonStars, IconWorld } from '@tabler/icons-react'
```
Comment (remadyt) [inline src/widget/header/ui/header.tsx]: add empty line

File: src/widget/header/ui/header.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                <Select
+                    leftSection={
+                        <IconWorld
+                            style={{ width: rem(18), height: rem(18) }}
+                            stroke={1.5}
+                        />
+                    }
+                    w={'50%'}
+                    defaultValue={'RU'}
+                    data={['RU', 'EN']}
```
Comment (remadyt) [inline src/widget/header/ui/header.tsx]: move to config

File: src/widget/header/ui/header.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        <IconSun
+            style={{ width: rem(16), height: rem(16) }}
+            stroke={2.5}
+            color={theme.colors.yellow[4]}
+        />
+    )
+
+    const moonIcon = (
+        <IconMoonStars
+            style={{ width: rem(16), height: rem(16) }}
```
Comment (remadyt) [inline src/widget/header/ui/header.tsx]: do not use inline style, here and all such places

**Review by remadyt — CHANGES_REQUESTED** [review]

File: src/widget/header/ui/header.tsx (hunk 1 above)
Comment (remadyt) [reply src/widget/header/ui/header.tsx]:
```
const themeIconMapper = {
  sun: IconSun,
  moon: IconMoon
};

const ThemeIcon = themeIconMapper[theme]

usage

<ThemeIcon ...all needed props here example={1.5} />

```

**Pull request merged**